### ISSUE TITLE
Не работи Ю при мене

### ISSUE THREAD

**Issue by boris-bc** [kristiyan-velkov/mac-os-bulgarian-phonetic-keyboard#4]
![image](https://github.com/kristiyan-velkov/mac-os-bulgarian-phonetic-keyboard/assets/63061368/15054533-a6c2-457c-8c52-fafc52d81322)

**Comment by boris-bc** [comment]
Случва се само с външната клавиатура

**Comment by kristiyan-velkov** [comment]
Това е много странно. Само тази буква ли е ?

На чт, 11.04.2024 г., 16:55 ч. Boris Petkov ***@***.***>
написа:

**Comment by boris-bc** [comment]
Да, мисля, че е само тази. Същото става и с други къстъм лейаути. Дефолтните лейаути, обаче, са си ОК.

**Comment by kristiyan-velkov** [comment]
Ще го погледна това със сигурност. Благодаря,че ми писахте.

На чт, 11.04.2024 г., 17:34 ч. Boris Petkov ***@***.***>
написа:

**Comment by boris-bc** [comment]
Пoдробност: клавиатурата на лаптопа е ISO, външната е US.

**Comment by kristiyan-velkov** [comment]
На лаптопа няма проблем?
Каква версия на Macos ползвате ?
Каква клавиатура модел ?

На чт, 11.04.2024 г., 17:49 ч. Boris Petkov ***@***.***>
написа:

**Comment by boris-bc** [comment]
Sonoma 14.4.1 (23E224)
[Ozone Tactical US](https://ozonegaming.com/en/collections/teclados/products/tactical-us)

Може би е свързано с клавиатурата, не е най-съвместимата на света

**Comment by borisdelev** [comment]
И при мен липсва на мясото на `. Мака е 15-inch, 2019, 14.5 ОС с тъч щуротията над клавиатурата (на неговата си, не външна).

**Comment by Eastforge** [comment]
Използвам Logitech MX Mechanical keyboard. Ю се визуализираше  на клавиша до Z бутона а не на § (първият бутон  от ляво). При мен реших проблема след като ресетнах Keyboard Type първоначалните настройки. Преминах от ANSI на ISO насттройка и всичко заработи както трябва.

**Comment by stoinov** [comment]
Потвърждавам, че промяната от ANSI към ISO оправи този проблем при мен.
За дебъг може да се ползва [Ukelele](https://software.sil.org/ukelele/), там при избор между двата типа в Settings, се визуализира същия проблем.

**Comment by kristiyan-velkov** [comment]
@stoinov, имаш ли идея как може да оправим този проблем генерално ?

**Comment by stoinov** [comment]
Почетох малко и явно е нещо такова:
McOS различава ANSI (нисък ентър) и ISO (висок ентър, допълнителен клавиш до левия Shift) клавиатурите и явно този клавиш е различен по някакъв начин в двете подредби.
В Ukelele промених подредбата да е правилна за ANSI, и сравних файловете:
редове 107 и 216 (code 50) бяха променени да имат съответно малко и главно Ю.

Премахнах стария метод за въвеждане и изтрих файла, рестрирах, сложих новия файл и пак рестартирах.
смених клавиатурата си да е ANSI, и пробвах да добавя променената подредба. Визуализира я правилно вече при избора:

<img width="534" height="330" alt="Image" src="https://github.com/user-attachments/assets/282cbf34-4a52-49f4-84a6-98360806bd7a" />

Изобразената клавиатура обаче все е ISO. След добавянето нямам визуализация както в първия коментар:

<img width="604" height="133" alt="Image" src="https://github.com/user-attachments/assets/7006ab9a-d430-46d7-a10f-079a561c9fef" />

За съжаление след това продължавам да не мога да пиша ю, ако не съм настроил да е ISO клавиатурата.

Намерих [следния проект](https://github.com/dilomo/NewBulgarianPhonetic_MacOS_KBLayout) с който успявам да пиша на ю както очаквам. Ползвам последния им релийз с бъндъл.
Не съм си играл да сравнявам файловете за разлики обаче.
